go/database/mpt/io/parallel_visit.go

```diff
@@ -420,6 +422,54 @@ func (f *stockNodeSourceFactory) open() (nodeSource, error) {
 	}, nil
 }
 
+// nodeSourceHashWithChildNodesFactory is a nodeSourceFactory implementation;
+// that creates stock backed sources to access nodes, using the node encoders
+// storing hashes of children with parent nodes.
+type nodeSourceHashWithChildNodesFactory struct {
+	directory string
+}
+
+func (f *nodeSourceHashWithChildNodesFactory) open() (nodeSource, error) {
```

Comment:
Instead of duplicating the code you could have parameterized the open code with an MptConfig and reuse the [getEncoder](https://github.com/Fantom-foundation/Carmen/blob/a7cc9b3cbcfef26ebd69d5ac3076258f3fcdac6d/go/database/mpt/forest.go#L1048) function to geth the encoder needed for the individual stocks. I think this would also make the `open` function itself more readable.

Reply:
ah, good idea!